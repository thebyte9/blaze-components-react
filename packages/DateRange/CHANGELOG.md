# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)

**Note:** Version bump only for package @blaze-react/date-range





# [0.8.0-alpha.3](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.2...v0.8.0-alpha.3) (2021-06-17)

**Note:** Version bump only for package @blaze-react/date-range





# [0.8.0-alpha.2](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.1...v0.8.0-alpha.2) (2021-06-17)

**Note:** Version bump only for package @blaze-react/date-range





# [0.8.0-alpha.1](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.0...v0.8.0-alpha.1) (2021-06-17)

**Note:** Version bump only for package @blaze-react/date-range





# [0.8.0-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.2-alpha.0...v0.8.0-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/date-range





## [0.6.2-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.1...v0.6.2-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/date-range
