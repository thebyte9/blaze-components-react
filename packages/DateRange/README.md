## Description

DateRange component allows you to select a data range among several options.

## Usage

```js
<DateRange onChange={({ startDate, endDate }) => {}} selected="custom" />
```

## API

##### DateRange can receive a number of `props` as follow:

| NAME     |       TYPE       |           DEFAULT            |
| :------- | :--------------: | :--------------------------: |
| selected | String or Number |            'any'             |
| onChange |     Function     | ({startDate, endDate}) => {} |
