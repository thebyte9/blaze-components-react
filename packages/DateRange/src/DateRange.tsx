import Select from '@blaze-react/select';
import React, { useState } from 'react';
import { ANY_DATE, CUSTOM_DATE, DEFAULT_OPTIONS } from './constants';
import DateRangeSelectDay from './DateRangeSelectDay';
import { DateUtilsSingleton } from './utils';

interface IDateRangeProps {
  children?: any;
  onChange: (args: IOnChangeArguments) => void;
  selected: 1 | 2 | 3 | 7 | 30 | 12 | 'custom' | 'any';
  custom?: false;
}
interface IOnChangeArguments {
  end?: string;
  start?: string;
  selectedDate?: string;
}

const DateRange: React.SFC<IDateRangeProps> = ({ onChange, custom }) => {
  const [selectedOption, setSelectedOption] = useState<string>(custom ? CUSTOM_DATE : '');
  const [selectedDate, setSelectedDate] = useState<Record<string, unknown>>({});

  const handleOnChange = ({ value }: { event: React.ChangeEvent<HTMLSelectElement>; value: string }) => {
    setSelectedOption(value);

    if (value !== CUSTOM_DATE && value !== ANY_DATE) {
      const [type, total] = value.split(',');
      const substractedDate = DateUtilsSingleton.subtractDate(Number(total), type);
      onChange({ selectedDate: substractedDate });
    }
  };

  const handleSelectedDate = (date: any) => {
    const rangeDate = {
      ...selectedDate,
      ...date,
    };
    setSelectedDate(rangeDate);
    onChange({ selectedDate: { ...rangeDate } });
  };

  const isCustom = selectedOption === CUSTOM_DATE;

  return (
    <>
      <Select label="Date Range" selected={selectedOption} options={DEFAULT_OPTIONS} onChange={handleOnChange} />
      {isCustom && (
        <>
          <DateRangeSelectDay onChange={handleSelectedDate} type="From" />
          <DateRangeSelectDay onChange={handleSelectedDate} type="To" />
        </>
      )}
    </>
  );
};

DateRange.defaultProps = {
  custom: false,
  onChange: () => void 0,
};

export default DateRange;
