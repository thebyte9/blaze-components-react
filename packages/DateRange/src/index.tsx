// @ts-nocheck
import Select from "@blaze-react/select";
import React, { useState } from "react";
import { IDateRangeProps } from "../interfaces";
import { ANY_DATE, CUSTOM_DATE, DEFAULT_OPTIONS } from "./constants";
import DateRangeSelectDay from "./DateRangeSelectDay";
import { DateUtils } from "./utils";

const DateRange: React.SFC<IDateRangeProps> = ({ onChange, custom }) => {
  const [selectedOption, setSelectedOption] = useState<string | number>(
    custom ? CUSTOM_DATE : ""
  );
  const [selectedDate, setSelectedDate] = useState<object>({});

  const handleOnChange = ({ value } = {}) => {
    setSelectedOption(value);

    if (value !== CUSTOM_DATE && value !== ANY_DATE) {
      const [type, total] = value.split(",");
      const substractedDate = DateUtils.subtractDate(total, type);
      onChange({ selectedDate: substractedDate });
    }
  };

  const handleSelectedDate = (date: any) => {
    const rangeDate = {
      ...selectedDate,
      ...date,
    };
    setSelectedDate(rangeDate);
    onChange({ selectedDate: { ...rangeDate } });
  };

  const isCustom = selectedOption === CUSTOM_DATE;

  return (
    <>
      <Select
        label="Date Range"
        selected={selectedOption}
        options={DEFAULT_OPTIONS}
        onChange={handleOnChange}
      />
      {isCustom && (
        <>
          <DateRangeSelectDay onChange={handleSelectedDate} type="From" />
          <DateRangeSelectDay onChange={handleSelectedDate} type="To" />
        </>
      )}
    </>
  );
};

DateRange.defaultProps = {
  custom: false,
  onChange: () => void 0,
};
export default DateRange;
