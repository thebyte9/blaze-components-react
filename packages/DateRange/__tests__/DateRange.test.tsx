import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import DateRange from '../src/DateRange';
import React from 'react';

const FIRST_DAY_OF_SECONND_ROW = 7;
const LAST_DAY_OF_SECONND_ROW = 13;
const LAST_30_DAYS = 30;

const defaultProps = (override: object = {}) => ({
  onChange: () => void 0,
  custom: true,
  ...override,
});

describe('DateRange component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const { asFragment } = render(<DateRange {...defaultProps({ selected: LAST_30_DAYS })} />);
    expect(asFragment()).toBeDefined();
  });

  test('should select a value', () => {
    render(<DateRange {...defaultProps()} />);
    fireEvent.change(screen.getByText('Please Choose...'), { target: { value: 'days, 30' } });
  });

  test('should select a value', () => {
    render(<DateRange {...defaultProps()} />);
    fireEvent.change(screen.getByText('Please Choose...'), { target: { value: 'days, 30' } });
  });
});
