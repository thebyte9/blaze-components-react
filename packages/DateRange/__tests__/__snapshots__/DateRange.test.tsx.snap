// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRange component should be defined and renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="form-field form-field--select"
  >
    <label
      class=""
    >
      Date Range
    </label>
    <select
      data-testid="select-date-range"
    >
      <option>
        Please Choose...
      </option>
      <option
        value="days,1"
      >
        Last 24 hours
      </option>
      <option
        value="days,2"
      >
        Last 48 hours
      </option>
      <option
        value="days,3"
      >
        Last 72 hours
      </option>
      <option
        value="days,7"
      >
        Last 7 days
      </option>
      <option
        value="days,30"
      >
        Last 30 days
      </option>
      <option
        value="years,1"
      >
        Last 12 months
      </option>
      <option
        value="custom"
      >
        Custom range date
      </option>
      <option
        value="any"
      >
        Any date
      </option>
    </select>
  </div>
  <div
    class="calendar-container"
    data-testid="calendar-container"
  >
    <div
      class="calendar-input"
    >
      <span
        class="calendar-input__label"
      >
        From
      </span>
      <div
        class="form-field form-field--input  "
      >
        <label
          class=""
        />
        <input
          class="range-date"
          data-testid="From"
          placeholder="dd/mm/yy"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    class="calendar-container"
    data-testid="calendar-container"
  >
    <div
      class="calendar-input"
    >
      <span
        class="calendar-input__label"
      >
        To
      </span>
      <div
        class="form-field form-field--input  "
      >
        <label
          class=""
        />
        <input
          class="range-date"
          data-testid="To"
          placeholder="dd/mm/yy"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
